router/http.go

```diff
@@ -491,9 +492,11 @@ func (s *httpService) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 				continue
 			}
 			log.Println("http: proxy error:", err)
+			outreq.Body.(*fakeCloseReadCloser).RealClose()
 			fail(w, 503)
 			return
 		}
+		outreq.Body.(*fakeCloseReadCloser).RealClose()
```

Comment:
I'm not super happy with the way this ends up being duplicated here, but didn't think it was worth the effort to try and factor out this retry loop in light of our other pending PR (#842). Maybe we should just make sure it's done more cleanly there?